src/procesos/principal.cpp

```diff
@@ -44,36 +51,50 @@ void MainProcess::run() {
     exit(0);
   }
 
-
-  std::vector<Jugador> v;
-  for (int i = 0; i < JUGADORES_PARA_TORNEO; ++i) {
-    v.push_back(Jugador(i));
+  this->pidTorneo = fork();
+  if (this->pidTorneo == 0) {
+    Torneo t(opts_);
+    t.run();
+    exit(0);
   }
 
-  // Si hay jugadores suficientes, lanzar torneo
-  // TODO: si no, esperar a que haya suficientes
-  if (v.size() >= JUGADORES_PARA_TORNEO) {
-    this->pidTorneo = fork();
-    if (this->pidTorneo == 0) {
-      Torneo t(v, opts_);
-      t.run();
-      exit(0);
-    } else {
-
-      int i = v.size();
-      while (i <= opts_.jugadores) {
-        milisleep(this->opts_.sleepJugadores);
-        Logger::getInstance()->info("[Principal] enviando señal SIGUSR1 al torneo: " + std::to_string(i));
-        kill(pidTorneo, SIGUSR1);
-        i++;
-      }
-      int status = 0;
+    this->agregadorDeJugadores = fork();
+    if (this->agregadorDeJugadores == 0) {
+        SIGINT_Handler sigint_handler;
+        SignalHandler :: getInstance()->registrarHandler (SIGINT, &sigint_handler);
 
-      wait(&status);
+        while (sigint_handler.getGracefulQuit() != 1) {
+            int resultado = semaforoEntradaJugadores.p();
+
+            if (resultado == -1) {
+                perror("[Principal] Posible error: ");
+                break;
+            }
+            int i = 0;
 
-      this->pidTorneo = 0;
+
+            milisleep(this->opts_.sleepJugadores);
+            Logger::getInstance()->info("[Principal] enviando señal SIGUSR1 al torneo: " + std::to_string(i));
```

Comment:
acá la definición del i y el to string está demás. Siempre loguea: "enviando señal SIGUSR1 al torneo: 0"